### ISSUE TITLE
Leaflet Tag Filter Button and LayerSupport Plugins

### ISSUE THREAD

**Issue by schmidtster** [ghybs/Leaflet.MarkerCluster.LayerSupport#15]
I noticed that this plugin has the capability to be compatible with other plugins that add or remove markers on a map. I was hoping to use this feature with the [Leaflet Tag Filter Button][1] made by maydemirx. It filters markers by associated tags and generates a little button for users to interact it. I believe I succeeded in adding the layerSupported cluster to my map,  but I am unsure, though, how to integrate the supported cluster with the tag filter buttons. The closest tutorial to combining two plugins is the Leaflet slider plugin, that has a demo here. I've tried to mimic the style of the implementation of the slider demo, adjusting for my own needs, but to no avail.

The Leaflet Tag Filter Button does support an [update() method and an enablePruneCluster method][4], both of which sound like they could be used to achieve what I'm looking for. Yet, when I apply them individually to the filter buttons, they don't work. I'm either applying the filter button methods incorrectly or creating the layerSupported cluster inaccurately.

Here is my code for generating the layer supported marker cluster group:

    var clusters = L.markerClusterGroup.layerSupport({maxClusterRadius:75}),
        group1 = L.layerGroup();
    var getjson = $.getJSON("map-v2.geojson",function(data){
      var bev = L.geoJson(data,{
        pointToLayer: function(feature,latlng){
          var marker = L.marker(latlng, { tags: feature.properties.Genres.concat(feature.properties.Creator)});
          marker.bindPopup('<p align=center>' + '<strong>Title: </strong>' + feature.properties.Title + '<br/><a href="' + feature.properties.Image_Bank_URL + '" target="_blank"><img src="' + feature.properties.Thumbnail_URL + '"/></a><br/>' + '<strong>Date: </strong>' + feature.properties.Date + '<br/>' + '<strong>Creator: </strong>' + feature.properties.Creator, {minWidth : 250});
          return marker;
        }
      });
      bev.addTo(group1);
      clusters.addLayer(group1);
      map.addLayer(clusters);
    });

    // Here is where I add the layer supported clusters to the map.
    clusters.checkIn(group1);
    clusters.addTo(map);

Here is the section where I generate the tag filter buttons:

    // Here is the code block for the Tag Filter Button. I start by accessing a tags file that has the data that I use for filter options. I should note that the genres.addToRelated is not working (it is supposed to link the 2 buttons together to work in conjunction with each other).
    $.getJSON('tags.json', function(data) {
    var genres = L.control.tagFilterButton({
      data: data.genres,
      filterOnEveryClick: true,
      icon: '<i class="fas fa-tags"></i>',
    }).addTo(map);
    var creators = L.control.tagFilterButton({
      data: data.creators,
      filterOnEveryClick: true,
      icon: '<i class="fas fa-user-edit"></i>',
    }).addTo(map);
    jQuery('.easy-button-button').click(function() {
		target = jQuery('.easy-button-button').not(this);
		target.parent().find('.tag-filter-tags-container').css({
			'display' : 'none',
		});
	});
    genres.addToRelated(creators);
    });

I added genres.update(clusters); and genres.enablePruneCluster(clusters); after the genres.addToRelated(creators);, but both of those didn't work to update the clusters.

If you'd like to see it all in action, here is a [plunker of the code][5].

  [1]: https://github.com/maydemirx/leaflet-tag-filter-button
  [3]: https://www.kent.ac.uk/parrotnet/pnetfiles/pnetbigdata.html
  [4]: https://github.com/maydemirx/leaflet-tag-filter-button#methods
  [5]: https://next.plnkr.co/edit/Ee8QYy?preview

**Comment by schmidtster** [comment]
This issue was solved over at Stack Overflow by ghybs: https://stackoverflow.com/questions/51770138/leaflet-tag-filter-button-and-markercluster-layersupport-plugins-integration-t

tl;dr: LayerSupport is not needed for integration. Changing the default functions of the Tag Filter Button's registerCustomSource and enabling MarkerClusterGroup as a layer to be filtered solves the issue.
